contributingGuides/TYPESCRIPT_STYLE.md

```diff
@@ -0,0 +1,857 @@
+
+// index.native.ts
+function isVisible() {
+    return AppState.currentState === 'active';
+}
+
+function hasFocus() {
+    return true;
+}
+
+const Visibility: VisibilityInterface = {
+    isVisible,
+    hasFocus,
+};
+
+export default Visibility;
+```
+
+Example with no default implementation:
+
+```ts
+// types.ts
+type Platform = 'android' | 'ios' | 'web' | 'desktop';
+export default Platform;
+
+// index.d.ts
+export default function getPlatform(): Platform;
+
+// index.ios.ts
+export default function getPlatform(): Platform {
+  return "ios";
+}
+
+// index.android.ts
+export default function getPlatform(): Platform {
+  return "android";
+}
+
+// index.desktop.ts
+export default function getPlatform(): Platform {
+  return "desktop";
+}
+
+// index.website.ts
+export default function getPlatform(): Platform {
+  return "website";
+}
+```
+
+Refer to the React Native documentation for more information about [Platform Specific Code](https://reactnative.dev/docs/platform-specific-code).
+
+## Naming Conventions
+
+Use **PascalCase** for class, type, interface, and enum names, as well as enum members and type parameters (generics). This improves readability and conforms to widely accepted TypeScript standards.
+
+
+```ts
+// Bad
+class someService {}
+type some_type = number | string;
+interface some_interface {}
+enum some_enum { some_value }
+function example<Some_type>(): Some_type {}
+
+// Good
+class SomeService {}
+type SomeType = number | string;
+interface SomeInterface {}
+enum SomeEnum { SomeValue }
+function example<SomeType>(): SomeType {}
+```
+
+Use **camelCase** for variables, parameters, functions, methods, properties, and module aliases.
+
+```ts
+// Bad
+const user_Name = "John";
+function process_Data(input_data: string) {}
+class BadExample {
+  do_Work() {}
+  WorkDone: boolean;
+}
+import * as Some_Module from "some-module";
+
+// Good
+const userName = "John";
+function processData(inputData: string) {}
+class GoodExample {
+  doWork() {}
+  workDone: boolean;
+}
+import * as someModule from "some-module";
+
+```
+
+Use **CONSTANT_CASE** for global constant values to distinguish them from other variables and emphasize their fixed state.
+```ts
+// Bad
+const someConst = { config: “value” } as const;
+
+// Good
+const SOME_CONST = { config: “value” } as const;
+```
+
+Avoid using common generic words such as data, state, amount, number, value, manager, engine, object, entity, instance, helper, util, broker, metadata, process, handle, and context.
+
+Especially do not include "Type" at the end of your type files and definitions. 
+ 
+```ts
+// Bad: ReportType.ts
+// Good: Report.ts
+
+// Bad
+type ReportType = {
+    // properties
+}
+
+// Good
+type Report = {
+    // properties
+}
+```
+
+Remove any words that are already clear from the variable's type declaration.
+
+```ts
+// Bad
+let nameString: string;
+let holidayDateList: Array<Date>;
+
+// Good
+let name: string;
+let holidays: Array<Date>;
```

Comment:
Is there any reason we are using `Array<Date>` instead of `Date[]`? 

* In my opinion `Date[]` makes more sense and is more widely understood as arrays are kinda like a primary type whereas `Array` could be misunderstood as a custom generic type.
* Also `Date[]` is less letters / faster to type
* `Date[5]` could be used for constant sized arrays

Reply:
Yes, we discussed about this in the document and we agreed about using  T[] for most cases (literals and easier types), and leaving Array<T> for more complex cases.

I will create a section about this.